File: leetcode/easy/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] output = new int[2];
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            if (hashMap.containsKey(target - nums[i])) {
                output[0] = hashMap.get(target - nums[i]);
                output[1] = i;
            }
            hashMap.put(nums[i], i);
        }
        return output;
    }
}

File: coderbyte/twoSum.js
function TwoSum(arr) { 
  let pairs = [];
  let dict = {};
  let target = arr[0];
  let output = '';

  for (let i = 0; i < arr.length; i++) {
    if ((target - arr[i]) in dict) {
      pairs.push([dict[target-arr[i]], i]);
    }
    dict[arr[i]] = i;
  }
  
  pairs.sort((a, b) => a[0] - b[0]);
  
  for (let [x, y] of pairs) {
    output += arr[x] + ',' + arr[y] + ' ';
  }

  return output.length ? output.trim() : -1;
}
